### DISCUSSION TITLE
I write a RSS reader for android based on slint-1.5.1, but I encounter few troubles. Any help is welcome.

### DISCUSSION THREAD

**Discussion by heng30** [slint-ui/slint#4985] [Q&A]
This is the project [rssbox-android](https://github.com/Heng30/rssbox-android).

- Can not copy and paste on LineEdit. Does any rust crate can help?
- Finish inputing some words to a LineEdit and transfering the focus to a FocusScope, which will hiding the keyboard. However, when I swith the app to the background and switch it back to the foreground. It would show the keyboard, that not I want. I think if I can unfocus the entire UI could fix this problem. Any help is welcome.
- Gestures is not support well. For example, Slide the screen from the right to left, which would trigger a `back` event. How to catch system events?
- The ListView is low efficiency on not fixed height child elements. The screen would jitter when I scroll, even with 20~30 child elements.  By the way, fixed height elements is fine, event with 200 child elements.

**Comment by ogoffart** [comment]
> Can not copy and paste on LineEdit. Does any rust crate can help?

This is implemented in the master branch.

> Finish inputing some words to a LineEdit and transfering the focus to a FocusScope, which will hiding the keyboard. However, when I swith the app to the background and switch it back to the foreground. It would show the keyboard, that not I want. I think if I can unfocus the entire UI could fix this problem. Any help is welcome.

That looks like a bug. I'd thought the keyboard would only be shown if a LineEdit has the focus.

> Gestures is not support well

Indeed, tnis is not yet implemented. 
You can maybe do something with TouchArea and check for mouse move

> The ListView is low efficiency on not fixed height child elements. The screen would jitter when I scroll, even with 20~30 child elements. 

What do you mean by jitter?
My intuition is that it is because it needs to recompute elements position within the ListView

**Reply by heng30** [reply]
> This is implemented in the master branch.

Thanks, I will try it.

> You can maybe do something with TouchArea and check for mouse move

For example, If the app on the second level, I move the finger from right screen edge to left. It would trigger a system back event. If I don't handle this event, that would cause the app exit. The normal mobile app will return back to the first level.  I think TouchArea can not handle the system events likes this. 

> My intuition is that it is because it needs to recompute elements position within the ListView

I agree with you. I think nothing I can do, unless touching with the source codes.
